R/dt_options.R

```diff
@@ -131,28 +167,107 @@ dt_options_init <- function(data) {
     "source_notes_border_lr_color",       TRUE,  "source_notes",     "value",   "#D3D3D3",
     "row_striping_background_color",      TRUE,  "row",              "value",   "rgba(128,128,128,0.05)",
     "row_striping_include_stub",         FALSE,  "row",              "logical", FALSE,
-    "row_striping_include_table_body",   FALSE,  "row",              "logical", TRUE,
-  )[-1, ] %>%
-    dt_options_set(options = ., data = data)
+    "row_striping_include_table_body",   FALSE,  "row",              "logical", FALSE,
+  )[-1, ]
+
+dt_options_get_default_value <- function(option) {
+
+  # Validate the provided `option` value
+  if (length(option) != 1) {
+    stop("A character vector of length one must be provided")
+  }
+  if (!(option %in% dt_options_tbl$parameter)) {
+    stop("The `option` provided is invalid")
+  }
+
+  dt_options_tbl$value[[which(dt_options_tbl$parameter == option)]]
 }
 
-dt_options_set_value <- function(data, option, value) {
+# Create an option-value list with a vector of arg names from the
+# `tab_options()` function and either one value or n-length values
+# corresponding to those options
+create_option_value_list <- function(tab_options_args, values) {
 
-  dt_options <-
-    data %>%
-    dt_options_get()
+  # Validate the length of the `values` vector
+  if (length(values) == 1) {
+    values <- rep(values, length(tab_options_args))
+  } else if (length(values) != length(tab_options_args)) {
+    stop("The length of the `values` vector must be 1 or the length of `tab_options_args`")
+  }
 
-  dt_options$value[[which(dt_options$parameter == option)]] <- value
+  # Validate the elements of the `tab_options_args` vector
+  validate_tab_options_args(tab_options_args)
 
-  dt_options %>%
-    dt_options_set(options = ., data = data)
+  stats::setNames(object = values, tab_options_args) %>% as.list()
 }
 
-dt_options_get_value <- function(data, option) {
+create_default_option_value_list <- function(tab_options_args) {
 
-  dt_options <-
-    data %>%
-    dt_options_get()
+  # Validate the elements of the `tab_options_args` vector
+  validate_tab_options_args(tab_options_args)
 
-  dt_options$value[[which(dt_options$parameter == option)]]
+  tab_options_args %>%
+    vapply(
+      FUN.VALUE = character(1),
+      FUN = function(x) {
+        stats::setNames(
+          object = dt_options_get_default_value(tidy_gsub(x, ".", "_", fixed = TRUE)),
+          x
+        )
+      }) %>%
+    as.list()
+}
+
+# Do multiple calls of `tab_options()` with an option-value list (`options`)
+tab_options_multi <- function(data, options) {
+  do.call(tab_options, c(list(data = data), options))
+}
+
+# Create vector of all args from `tab_options()` that involve line styles
+table_line_style_vec <- function() {
```

Comment:
Let's introduce a function that does `formals(tab_options) %>% names() %>% setdiff("data") %>% grep(x, ., value = TRUE)`

Reply:
Added `get_tab_options_arg_vec()`. Thanks!